model/edx.go

```diff
@@ -67,6 +67,45 @@ func (e *EdxLMS) GetOAuthConsumerSecret() string {
 	return e.OAuthConsumerSecret
 }
 
+func (e *EdxLMS) GetTeams() map[string]string {
+	return e.Teams
+}
+
+func (e *EdxLMS) CheckIfActiveLMS(currentTeamName string) bool {
+	// For every property of Teams in this learning management service (LMS) object (passed as e)
+	// The team's slug names are the values of each key, value pair
+	// Represent each team's slug name with teamSlug, _ as an empty placeholder for the key of this lms object
+	for _, teamSlug := range e.GetTeams() {
+		// If this team slug is equal to the team slug of the currently selected mattermost team in the web app
+		if teamSlug == currentTeamName {
+			return true
+		}
+	}
+	return false
+}
+
+func (e *EdxLMS) GetLearningGroupPrefixes() ([]*LearningGroupPrefix, *AppError) {
+	var personalChannelNames []*LearningGroupPrefix = make([]*LearningGroupPrefix, len(e.PersonalChannels.ChannelList))
+
+	// Loop over PersonalChannels.ChannelList in the active LMS configuration and, for every key,
```

Comment:
This is better, but I'd still rather see a description of _why_ we're doing this than a description of _what_ we're doing here. The what is fairly easy to understand from the code, but the why has to be guessed at.

Reply:
Ok I added another line in the comment, let me know if that is the 'why' you are looking for

Reply:
Also, now that I've learned a little more Go, could you not write:

`for personalChannelName := range e.PersonalChannels.ChannelList {
`     append(personalChannelNames, &LearningGroupPrefix{Prefix: personalChannelName})`
`}`

That's cleaner and save the local variable. If you like, you can also then omit the `make(...)`  call on line 88 because go's append will automatically handle amortized growth.

Reply:
> Ok I added another line in the comment, let me know if that is the 'why' you are looking for

Closer. I've dug in a little more now and I think I understand what this is for. How about:

"Construct an slice containing pointers to the prefixes of the names of each channel that this user is a part of."

I _think_ that's what we're really trying to do here.

Reply:
Your not really getting anything for this specific user, but I like where you went with. I altered it to:

> Construct a slice containing pointers to the prefixes of the names of private channels that are for learning groups in this LMS

Reply:
That's exactly the kind of info I'm looking for!

Reply:
I don't think a comment describing the type of data such as "slice containing pointers" is a particularly good comment because it's a comment stating exactly what the code does.

Comments in the code can explain unusual constructs, or broad steps of the algorithm being used to accomplish what the function is supposed to accomplish
```
// Find the LMS for the active team
...
// Get the list of personal channel names from that LMS because those
// are the prefixes used for learning groups
...
// Create a learning group type for each channel name setting the prefix to that name
...
```
things like that. Where each of the `...` 's is the block of code to accomplish that comment.

Reply:
@mlippert I like the idea of dropping the "slice containing pointers". Your documentation is significantly nicer for this.